petab/conditions.py

```diff
@@ -1,35 +1,54 @@
 """Functions operating on the PEtab condition table"""
 
-from typing import Iterable, Optional, List
+from typing import Iterable, Optional, List, Union
 
 import numpy as np
 import pandas as pd
 
 from . import lint, parameters, core
+from .C import *
 
 
-def get_condition_df(condition_file_name: str) -> pd.DataFrame:
+def get_condition_df(
+        condition_file_name: Union[str, pd.DataFrame, None]
+) -> pd.DataFrame:
     """Read the provided condition file into a ``pandas.Dataframe``
 
     Conditions are rows, parameters are columns, conditionId is index.
 
     Arguments:
         condition_file_name: File name of PEtab condition file
     """
+    if condition_file_name is None:
+        return condition_file_name
+
+    if isinstance(condition_file_name, pd.DataFrame):
+        return condition_file_name
 
     condition_df = pd.read_csv(condition_file_name, sep='\t')
     lint.assert_no_leading_trailing_whitespace(
         condition_df.columns.values, "condition")
 
     try:
-        condition_df.set_index(['conditionId'], inplace=True)
+        condition_df.set_index([CONDITION_ID], inplace=True)
     except KeyError:
         raise KeyError(
-            'Condition table missing mandatory field `conditionId`.')
+            f'Condition table missing mandatory field {CONDITION_ID}.')
 
     return condition_df
 
 
+def write_condition_df(df: pd.DataFrame, filename: str) -> None:
```

Comment:
is this not generic for all tsv files?

Reply:
For some we want to set the right index column, for others we won't. But yes, some are same. Would still keep it separate in case something changes. One might for example include some basic checking there at some point.